src/librustc/mir/mod.rs

```diff
@@ -147,7 +147,11 @@ pub struct Mir<'tcx> {
 
     /// Names and capture modes of all the closure upvars, assuming
     /// the first argument is either the closure or a reference to it.
-    pub upvar_decls: Vec<UpvarDecl>,
+    // NOTE(eddyb) This is *strictly* a temporary hack for codegen
+    // debuginfo generation, and will be removed at some point.
+    // Do **NOT** use it for anything else, upvar information should not be
+    // in the MIR, please rely on local crate HIR or other side-channels.
+    pub __upvar_debuginfo_codegen_only_do_not_use: Vec<UpvarDebuginfo>,
```

Comment:
lol ok I guess that addresses my earlier concern

Reply:
I would like this to be part of the doc comment please though, why use only `//`?

Reply:
I got used to only using `//` for NOTE/FIXME/HACK, but I'll go change it.